pkg/repo/scanner.go

```diff
@@ -76,23 +116,118 @@ func Scan(dir string, diffCmd []string, platformUrl string, consoleUrl string, v
 		return fmt.Errorf("failed to get presigned URL: %w", err)
 	}
 
+	fmt.Printf("Uploading package repo...\n")
+
 	if err := uploadFileToS3(presignedUrl, filepath.Join(tarballDir, tarballName)); err != nil {
 		return fmt.Errorf("failed to upload file to S3: %w", err)
 	}
 
-	epoch, err := url.GetEpochFromUrl(presignedUrl)
+	epoch, err := urlBuilder.GetEpochFromUrl(presignedUrl)
 	if err != nil {
 		return err
 	}
 
-	consoleFullUrl, err := url.Build(consoleUrl, "analysis/users", *epoch, "result")
+	consoleFullUrl, err := urlBuilder.Build(consoleUrl, "analysis/users", *epoch, "result")
 	if err != nil {
 		return err
 	}
 
-	fmt.Printf("Success, your scan is processing! Once completed, you can see results here: %s\n", *consoleFullUrl)
+	fmt.Printf("Upload successful, your scan is processing!\n")
 
-	return nil
+	// At the beginning of your polling function
```

Comment:
Can we move all this to a new function, and keep the top-level `Scan()` small?

Reply:
done